fineract-core/src/main/java/org/apache/fineract/infrastructure/event/external/jobs/SendAsynchronousEventsTasklet.java

```diff
@@ -104,16 +109,34 @@ private void markEventsAsSent(List<Long> eventIds) {
         // Partitioning dataset to avoid exception: PreparedStatement can have at most 65,535 parameters
         final int partitionSize = fineractProperties.getEvents().getExternal().getPartitionSize();
         List<List<Long>> partitions = Lists.partition(eventIds, partitionSize);
+        final ThreadPoolTaskExecutor executor = getThreadPoolTaskExecutor();
+        final FineractContext context = ThreadLocalContextUtil.getContext();
         partitions //
                 .forEach(partitionedEventIds -> {
-                    measure(() -> {
-                        repository.markEventsSent(partitionedEventIds, sentAt);
-                    }, timeTaken -> {
-                        log.debug("Took {}ms to update {} events", timeTaken.toMillis(), partitionedEventIds.size());
+                    executor.execute(() -> {
+                        measure(() -> {
+                            ThreadLocalContextUtil.init(context);
+                            repository.markEventsSent(partitionedEventIds, sentAt);
+                        }, timeTaken -> {
+                            log.debug("Took {}ms to update {} events", timeTaken.toMillis(), partitionedEventIds.size());
+                        });
                     });
                 });
     }
 
+    @Nonnull
+    private ThreadPoolTaskExecutor getThreadPoolTaskExecutor() {
```

Comment:
Don't create this inside the Tasklet. Extract it to the tasklet's config (that's why it's there) and have the threadpool configured there as a bean.

Reply:
Couldn't move it to the tasklet config as it would create a circular dependency between them, moved it to it's own config.